### PR TITLE
Nonlazy conv

### PR DESCRIPTION
**Pull request by samholtzen**

### CODE REVIEW

**Comment by samholtzen** [comment]
Training loop is working, and eval is working again after a problem with channels orientation.

The F1 dropped from 0.84 to 0.79, so I'm going to be looking into that. The drop in accuracy is most likely an issue with the MESS of changes I made before being enlightened in how badly I was using git :C

I think switching over to channels first is going to be my first order of business, then retraining the model with preprocessing set as channels first.

**Comment by samholtzen** [comment]
This final commit has cleared the backlog of changes. The comprehensive notes of what has changed:

1. Everything is now nucleus first, cytoplasm second! That means input data, training, eval, inference is always taken in and returned nucleus first, cytoplasm second.
2. Everything is now channels first (that is, `B, C, H, W`). This comes with API changes, where there is no longer a keyword argument for `data_format`. If you want to segment your images, please use `np.moveaxis` to change axis orientation from channels last to channels first.
3. Postprocessing now does a 1 px expansion of the interior prediction to account for the erosion done during training.
4. Changes to scripts reflecting the above changes.
5. The DataLoader `__getitem__` function now features a pretty hacky workaround to solve the background padding issues. Originally, the padding for the background channel was 0, which caused poor training since it was incongruous with the reality of padding. Lines 135-136 manually edit the background channels for the nucleus and cytosol. TODO: a more elegant solution for this problem, since we're going to be incorporating capabilities for DynamicNuclearNet.

**Comment by samholtzen** [comment]
Newest push includes:
- changes to the preprocess script to save the `.npz` file to a Zarr including important metadata.
- The above change required rewriting the `loader.py`, `eval.py` and `train.py` scripts that added an extra key when indexing the zarr.

**Review by rossbar — APPROVED** [review]
Thanks so much @samholtzen , let's get this in and keep crankin!
